### PR TITLE
8295809: TreeTableViewSkin: memory leak when changing skin

### PR DESCRIPTION
**Pull request by andy-goryachev-oracle**
as determined by SkinMemoryLeakTest (remove line 180) and a leak tester
https://github.com/andy-goryachev-oracle/Test/blob/main/src/goryachev/apps/LeakTest.java

caused by:
- adding and not removing listeners
- adding and not removing event handlers/filters
- adding and not removing cell factory
- holding unnecessary instance in TreeTableViewSkin.treeTableViewSkin // test-only

NOTES:
1. this fix requires JDK-8294809 ListenerHelper and JDK-8295806 TableViewSkin.
2. there were more memory leaks in this skin than detected by SkinMemoryLeakTest, probably because of cell factory (?).  A manual test using LeakTest shows that there are no memory leaks after a) replacing skin, b) moving the components to a new window, and c) removing all the components from the window.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8295809](https://bugs.openjdk.org/browse/JDK-8295809): TreeTableViewSkin: memory leak when changing skin

### Reviewers
 * [Ajit Ghaisas](https://openjdk.org/census#aghaisas) (@aghaisas - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx pull/931/head:pull/931` \
`$ git checkout pull/931`

Update a local copy of the PR: \
`$ git checkout pull/931` \
`$ git pull https://git.openjdk.org/jfx pull/931/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 931`

View PR using the GUI difftool: \
`$ git pr show -t 931`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/931.diff">https://git.openjdk.org/jfx/pull/931.diff</a>

</details>

### CODE REVIEW

**Comment by andy-goryachev-oracle** [comment]
must be reviewed after https://github.com/openjdk/jfx/pull/929

**Comment by andy-goryachev-oracle** [comment]
@aghaisas could you please take a look at this PR?

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/SkinMemoryLeakTest.java [hunk 1]
```diff
@@ -232,10 +232,10 @@ public static Collection<Object[]> data() {
                 //MenuBar.class,
 
                 //
-                PasswordField.class,
+                //PasswordField.class,
```
Comment (aghaisas) [inline modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/SkinMemoryLeakTest.java:235]: This seems an unrelated change.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/SkinMemoryLeakTest.java [hunk 2]
```diff
@@ -232,10 +232,10 @@ public static Collection<Object[]> data() {
...
 
                 //
-                Spinner.class,
+                Spinner.class
```
Comment (aghaisas) [inline modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/SkinMemoryLeakTest.java:238]: This seems an unrelated change.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/SkinCleanupTest.java [hunk 3]
```diff
@@ -274,6 +274,8 @@ public void testTreeTableRowFixedCellSizeEnabled() {
         assertTrue("fixed cell size enabled", isFixedCellSizeEnabled(tableRow));
     }
 
+    @Ignore("JDK-8295809") // TODO probably need to verify the result of listener action,
```
Comment (aghaisas) [inline modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/SkinCleanupTest.java:277]: We need to file a follow-on bug to modify or cleanup these ignored tests in future.
Use the bug ID of the newly filed bug with `@Ignore` tag for these two tests.

**Review by aghaisas — APPROVED** [review]

**Review by aghaisas — CHANGES_REQUESTED** [review]
I have identified the minor changes required.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/SkinMemoryLeakTest.java (hunk 1 above)
Comment (andy-goryachev-oracle) [reply modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/SkinMemoryLeakTest.java]: since this is the last PR in the series of skin memory leak fixes, the code block will be removed.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/SkinMemoryLeakTest.java (hunk 2 above)
Comment (andy-goryachev-oracle) [reply modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/SkinMemoryLeakTest.java]: cleaned up.

**Review by aghaisas — APPROVED** [review]
Thanks for fixing the tests. Fix looks good now.

This will have some conflict with https://github.com/openjdk/jfx/pull/805 fix though.
